src/NumericalAlgorithms/Spectral/Spectral.hpp

```diff
@@ -419,6 +419,47 @@ template <typename T>
 Matrix interpolation_matrix(const Mesh<1>& mesh,
                             const T& target_points) noexcept;
 
+// @{
+/*!
+ * \brief Matrices that interpolate to the lower and upper boundaries of the
+ * element.
+ *
+ * Assumes that the logical coordinates are \f$[-1, 1]\f$. The first element of
+ * the pair interpolates to \f$\xi=-1\f$ and the second to \f$\xi=1\f$. These
+ * are just the Lagrange interpolating polynomials evaluated at \f$\xi=\pm1\f$.
+ * For Gauss-Lobatto points the only non-zero element is at the boundaries
+ * and is one and so is not implemented.
+ *
+ * \warning This can only be called with Gauss points.
+ */
+const std::pair<Matrix, Matrix>& boundary_interpolation_matrix(
+    const Mesh<1>& mesh) noexcept;
+
+template <Basis BasisType, Quadrature QuadratureType>
+const std::pair<Matrix, Matrix>& boundary_interpolation_matrix(
+    size_t num_points) noexcept;
+// @}
+
+// @{
+/*!
+ * \brief `DataVector`s of the Lagrange polynomials evaluated at the lower and
+ * upper boundaries divided by the volume quadrature weights.
```

Comment:
- I would find it more helpful if the brief gave me a description of what this is for, rather than just be a math formula transcribed to english. You have this information in the main paragraph, so perhaps that sentence could be moved up to become the brief? For instance, something like "The terms associated with lifting a boundary term into the volume for a discontinuous Galerkin scheme that uses Gauss points", or similar.
- The information in this brief is still good to have, so it moved into the main paragraph. Even better, the next paragraph could include the formula for the term, instead of describing the formula.
- I think an additional paragraph should be added to clarify how this relates to the flux lifting action, which brings in terms of its own, and how the two are to be combined in practice.

Reply:
I agree with the first two, I disagree with the third. I don't think this is the place to describe how to do DG. It's also not clear what to say that hasn't been said that wouldn't involve writing out the full DG expression. Could you clarify?

Reply:
My concern is that it will be confusing to non-experts to find both the `lift_flux` function that says it lifts into the volume (with no reference to this function), and this `boundary_lifting_term` function that says it's the weight for lifting (with no reference to the function that does the lifting).

But I do agree with your reservations, and I'm not sure how to resolve this. Perhaps a solution would be to have a central description of the DG scheme that points to both `lift_flux` and `boundary_lifting_term` in context of the overall discussion?

Reply:
Ah I see! So I am planning on adding a `lift_flux_gauss_points` function and rename the current one to `lift_flux_gauss_lobatto_points`, or have the necessary if-else logic inside the `lift_flux` function. I haven't decided which is cleaner. Either way, the two functions would be in the same file and reference each other in the documentation. Since the whole lifting operation in multi-d is fairly complicated.

I agree, we need a central discussion about DG. I have this pretty much written from my thesis, but haven't prioritized getting it into the dox.

Reply:
Oh okay, that clarifies things a lot. I think this will become sufficiently clear as soon as all the pieces are in, and then will only get better when we eventually get around to fully documenting the scheme. I'm satisfied!

Reply:
Okay! Thanks for bringing this up :) I think it's an important point to address :)